pelican/tests/test_readers.py

```diff
@@ -273,6 +273,37 @@ def test_article_extra_path_metadata_dont_overwrite(self):
         }
         self.assertDictHasSubset(page.metadata, expected)
 
+    def test_article_extra_path_metadata_recurse(self):
+        parent = "TestCategory"
+        notparent = "TestCategory/article"
+        path = "TestCategory/article_without_category.rst"
+
+        epm = {
+            parent: {'epmr_inherit': parent,
+                     'epmr_override': parent, },
+            notparent: {'epmr_bogus': notparent},
+            path:   {'epmr_override': path, },
+            }
+        expected_metadata = {
+            'epmr_inherit': parent,
+            'epmr_override': path,
+            }
+
+        page = self.read_file(path=path, EXTRA_PATH_METADATA=epm)
+        self.assertDictHasSubset(page.metadata, expected_metadata)
+
+        # Make sure vars aren't getting "inherited" by mistake...
+        path = "article.rst"
+        page = self.read_file(path=path, EXTRA_PATH_METADATA=epm)
+        for k in expected_metadata.keys():
+            self.assertNotIn(k, page.metadata)
+
+        # Same, but for edge cases where one file's name is a prefix of
+        # another.
+        page = self.read_file(path=path, EXTRA_PATH_METADATA=epm)
```

Comment:
I think you want to test this with `path = "TestCategory/article_without_category.rst"` as defined at the beginning of this test, but you've redefined  `path` to `"article.rst"` in the previous block.

Reply:
Oops, yes, thanks. Fixed.